pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -283,6 +299,33 @@ func (gce *GCECloud) ProviderName() string {
 	return ProviderName
 }
 
+// Region returns the region
+func (gce *GCECloud) Region() string {
+	return gce.region
+}
+
+// OnXPN returns whether bool indicating whether the cluster is
+// running on a cross project network.
+func (gce *GCECloud) OnXPN() bool {
+	return gce.onXPN
+}
+
+// NetworkURL returns the network url
+func (gce *GCECloud) NetworkURL() string {
+	return gce.networkURL
+}
+
+// SubnetworkURL returns the subnetwork url
+func (gce *GCECloud) SubnetworkURL() string {
+	return gce.subnetworkURL
+}
+
+// NodeTags returns the list of node tags specified from the config file
+// If this list is empty, users should call GCECloud.ComputeHostTags
```

Comment:
Good question. 
When they're specified by config, we absolutely know that these are the tags to use. 
When this isn't set, we don't know what tags to use, so we need to calculate it. However, nodes could be added to the cluster later, so we need to re-generate tags when the set of hostnames has changed.  I've been planning to modify the ingress controller to check if the hostname set has changed - if not, use known value, else call Compute...

I can implement this feature within the cloud provider. I opted not to because I wasn't sure if users would want to know the difference - calculated vs config given.